SteamTogether.Bot/Program.cs

```diff
@@ -25,6 +26,51 @@
                 .AddCommandLine(args);
         }
     )
+    .ConfigureLogging(
+        (context, logging) =>
+        {
+            logging.ClearProviders();
+            logging.AddProvider(
+                new FilteredConsoleLoggerProvider(
+                    (category, level) =>
```

Comment:
This levels should not be hardcoded, it can be configured from the appsettings file
We can define default level as Warning and set other need logging as Information
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-7.0#configure-logging

Reply:
Okay, I see the problem maybe RedactLoggedHeaders method for TelegramBotClient can help here, but will need to check

Reply:
I think it won't help. In our case the URL gets logged.

Reply:
This should do: [Create new Custom ConsoleLogFormatter](https://learn.microsoft.com/en-us/dotnet/core/extensions/console-log-formatter#implement-a-custom-formatter)
But this will affect all console output and change the appearens, in our case I would just change the logging level by default to something like:

// appsettings.bot.json
```json
{ 
 "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft.Hosting.Lifetime": "Information",
      "System.Net.Http.HttpClient.TelegramBotClient": "Warning"
    }
}

Also possible to check out third-party libraries like log4net, serilog and etc, it might be easier to implement replace sensitive log messages

Reply:
> this will affect all console output and change the appearens

That might be not a bad thing. This way we'll make sure that the token won't appear in any logs.

> just change the logging level

I'm inclined to do that now as a quick mitigation, but this won't fix the root cause.

It is quite surprising that no straightforward ways to mutate log output can be googled. In my experience, it is required not often but from time to time.

> check out third-party libraries

That'd be fine for me if you pick something. I have little experience in C# and limited knowledge of the ecosystem.

Reply:
Sure, I'll take a look at 3rd party libs